pkg/ebpf/tracer.go

```diff
@@ -832,3 +836,64 @@ func (t *Tracer) getProbeProgramIDs() (map[string]uint32, error) {
 	}
 	return fds, nil
 }
+
+func connExistsFn(procRoot string) (func(*ConnTuple) bool, func()) {
```

Comment:
What do you think about moving this to our `netlink` package?

Reply:
Yeah I think we can do that. Let me give that a shot.

Reply:
Ok, so I attempted to move most of this to `netlink`, but I think the code is uglier and more inefficient, so I think it is best this remains as it is now.
1. I don't want to use `ConnTuple` in `netlink` because that would add a dependency on `ebpf` from `netlink`. Currently we have `ebpf -> network`, `ebpf -> netlink`, `netlink -> network`. Adding `netlink -> ebpf` would introduce a cyclic dependency.
2. Without using `ConnTuple` in `netlink`, it is not straightforward to check if we need a new `Conntrack` object for a network namespace. We have the inode number in `ConnTuple`, but we can't go from that to a fd that is required to create the netlink socket. This is why we are using the pid from `ConnTuple` below to get an fd to the network namespace. So any abstraction in `netlink` for this would need to have the pid, procRoot and `Con` object passed in. It doesn't result in a clean, well defined package boundary.

Reply:
gotcha. err, I have hit this issue before. I think we should probably move `ConnTuple` to a a separate package at some point, but it makes sense to leave this to another PR